XTS/WS-T/dev/src/org/jboss/jbossts/xts/recovery/participant/at/XTSATRecoveryManager.java

```diff
@@ -24,11 +24,19 @@ public abstract class XTSATRecoveryManager {
      */
     public static XTSATRecoveryManager getRecoveryManager()
     {
+        long currentTimeout = 1000 + System.currentTimeMillis() ;
+
         // use a lock to get the recovery manager to avoid a race with the code that sets it
         synchronized (lock) {
             // theRecoveryManager must eventually be set to a non-null value
             while (theRecoveryManager == null) {
                 try {
+
+                    if (currentTimeout - System.currentTimeMillis() < 0)
+                    {
+                        break ;
```

Comment:
This change means that a null value could be returned but the method is designed to guarantee that a non-null value is returned so other things will break.

Something must eventually call setRecoveryManager() which will call notifyAll causing this thread to return from the call to lock.wait(); and the loop condition will be re-evaluated.

The thread synchronisation is enforced by running the relevant code inside a synchronized (lock) code block.